Assignment 3.Rmd

```diff
@@ -147,9 +150,68 @@ Look at the number of people in each cluster, now repeat this process for 3 rath
 
 Using the data collected for Assignment 2 (which classes students were in), cluster the students, then redraw the graph of the class but color the students according the cluster they are in.
 
+```{r}
+library(tidyr)
+library(dplyr)
+library(klaR)
+H1 <- read.csv("C:/Users/Yiyi Xie/Documents/assignment2/hudk4050-classes.csv", header = TRUE)
+H1<- unite(H1,Student_Name,"First.Name","Last.Name",sep=" ")
+H1 <- filter(H1,Student_Name!= "ZIMO CHEN")
+H2 <- H1 %>%gather(course_number,course,Class.1:Class.6) %>% filter(course !="") %>% dplyr::select(Student_Name,course)
+H2$course= gsub(' ', '', H2$course)
+H2 <- H2 %>% filter(course != "HUDK4050")
+
+H2$Count <- 1
+H3 <- tidyr::spread(H2, course, Count)
+row.names(H3) <- H3$Name
+H3$Name <- NULL
+H3 <- ifelse(is.na(H3), 0, 1)
+H4 <- as.matrix(H3)
+H5 <- H4 %*% t(H4)
+diag(H5) <- NA
+library(klaR)
+fit <- kmodes(H3, 5, iter.max = 10, weighted = FALSE)
+fit$cluster
+H3a = data.frame(H3, fit$cluster)
+library(igraph)
+H <- graph.adjacency(H5,mode="undirected")
+plot(H,layout=layout.fruchterman.reingold, vertex.size=10, vertex.color=H3a$fit.cluster)
+
+```
+
 ##Part III
 
 In class activity 6 you clustered students in the class by the answers to a questionaire. Create a visualization that shows the overlap between these clusters and the clusters generated in part II.
+```{r}
+chooseCRANmirror(graphics=FALSE, ind=1)
+DF1 <-read.csv("C:/Users/Yiyi Xie/Documents/class-activity-6/HUDK405019-clustering.csv")
+DF1 <- unite(DF1, "Name", c("First.Name", "Last.Name"), sep = ".")
+row.names(DF1) <- DF1$Name
+DF1$Name <- NULL
+DF2 <- DF1[1:11]
+DF2 <- DF2 %>% mutate_all(funs(gsub("[a-zA-Z]", "", .)))
+DF2 <- DF2 %>% mutate_all(funs(as.numeric(.)))
+DF2 <- as.data.frame(scale(DF2))
+DF2 <- DF2 %>% mutate_all(funs(ifelse(is.na(.) == TRUE, 0, .)))
+
+install.packages("ggmap")
+install.packages("rgdal")
+library(ggmap)
+library(tmaptools)
+DF3 <- DF1[13:14]
+names(DF3) <- c("lattitude", "longitude")
+DF3 <- DF3 %>% mutate_all(funs(gsub("[a-zA-Z]", "", .)))
+DF3 <- DF3 %>% mutate_all(funs(sub("[?]", "", .)))
+DF3$lattitude <- sub(",.*$","", DF3$lattitude) 
+DF3$lattitude <- sub("°.*$","", DF3$lattitude)
+DF3$longitude <- gsub(".*,","",DF3$longitude)
+DF3$longitude <- sub("°.*$","", DF3$longitude)
+DF3 <- DF3 %>% mutate_all(funs(as.numeric(.)))
+fit <- kmeans(DF2, 3) 
+DF4 <- data.frame(DF2, DF3, fit$cluster)
+ggplot(DF4, aes(longitude, lattitude, color = as.factor(fit.cluster))) + geom_point(size = 5)
+
+```
```

Comment:
Overall, great work! You are on the right track but you might want to work on the analysis part of this assignment.